server/test/integration/vitalsTest/handleHeartbeatTest.js

```diff
@@ -537,4 +552,38 @@ describe('vitals.js integration tests: handleHeartbeat', () => {
       expect(helpers.logSentry).not.to.be.called
     })
   })
+
+  describe('This heartbeat indicates that door events are frequently missed', () => {
+    beforeEach(async () => {
+      await db.clearTables()
+
+      this.client = await factories.clientDBFactory(db)
+      await locationDBFactory(db, {
+        locationid: testLocation1Id,
+        radarCoreId: radar_coreID,
+        clientId: this.client.id,
+      })
+
+      await sensorsVitalDBFactory(db, {
+        locationid: testLocation1Id,
+        isTampered: true,
+      })
+
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(helpers, 'logSentry')
+      sandbox.stub(db, 'logSensorsVital')
+
+      await doorMissedFrequentlyHeartbeat(radar_coreID)
+    })
+
+    afterEach(async () => {
+      await db.clearTables()
+      sandbox.restore()
+    })
+
+    it('should log to Sentry that sensor is frequently missing door events', async () => {
+      expect(helpers.logSentry).to.be.calledWith(`Sensor is frequently missing door events at ${testLocation1Id}`)
+    })
```

Comment:
Looks good!

How did you find updating and adding to these test cases? Have you done much automated testing before? Do you have any questions about it?

Reply:
I found it fun and somewhat familiar! I've worked with automated testing before, primarily when working on the website that the Computing Science Student Society uses at SFU; our webadmin is particularly keen on having rigourous testing for any interaction with the API.